Source/WarpX.cpp

```diff
@@ -108,20 +108,20 @@ Real WarpX::quantum_xi_c2 = PhysConst::xi_c2;
 Real WarpX::gamma_boost = 1._rt;
 Real WarpX::beta_boost = 0._rt;
 Vector<int> WarpX::boost_direction = {0,0,0};
-int WarpX::do_compute_max_step_from_zmax = 0;
+bool WarpX::do_compute_max_step_from_zmax = false;
 Real WarpX::zmax_plasma_to_compute_max_step = 0._rt;
 
-long WarpX::current_deposition_algo;
-long WarpX::charge_deposition_algo;
-long WarpX::field_gathering_algo;
-long WarpX::particle_pusher_algo;
-int WarpX::maxwell_solver_id;
-long WarpX::load_balance_costs_update_algo;
+short WarpX::current_deposition_algo;
+short WarpX::charge_deposition_algo;
+short WarpX::field_gathering_algo;
+short WarpX::particle_pusher_algo;
+short WarpX::maxwell_solver_id;
+short WarpX::load_balance_costs_update_algo;
 bool WarpX::do_dive_cleaning = 0;
 bool WarpX::do_divb_cleaning = 0;
```

Comment:
I would also replace those with `true`/`false` values and tell other developers about it in the development meeting.

Writing `0`/`1` is C style and confusing to new developers with a non-C background.

We should also talk in the dev meeting about extending the parser to support named boolean values, similar to CMake `if`:
https://cmake.org/cmake/help/latest/command/if.html#basic-expressions

basically case-insensitive support for True/on and False/off.

Reply:
Thank you, Axel. I replaced those and the corresponding PML flags as well in https://github.com/ECP-WarpX/WarpX/pull/2922/commits/6d26570b1bd261822f07cd84856658ce03fafccb.